pkg/kubelet/apis/podresources/server_v1_test.go

```diff
@@ -276,6 +277,10 @@ func TestAllocatableResources(t *testing.T) {
 			if err != nil {
 				t.Errorf("want err = %v, got %q", nil, err)
 			}
+
+			sort.Slice(resp.CpuIds, func(i, j int) bool { return resp.CpuIds[i] < resp.CpuIds[j] })
+			sort.Slice(resp.Devices, func(i, j int) bool { return resp.Devices[i].ResourceName < resp.Devices[j].ResourceName })
```

Comment:
not good enough. This has still room for false negatives. Needs another update.

Reply:
Yes, as we are just sorting by resource name in case of devices if the resource name is same it we can't be sure of the order in which we have numa nodes  which could lead to mismatch because of order
eg example.com/deviceA on numa node 0 and example.com/deviceA on numa node 1 we can't sure on the way they are sorted and hence could still return a false negative when we compare the expected and obtained response strings.